src/replica/ReplicationTask.h

```diff
@@ -50,54 +49,40 @@ class ReplicationTask : public Task {
      * and memory management of instances created otherwise (as values or via
      * low-level pointers).
      *
-     * @param controller
-     *   a reference to the Controller for launching requests, jobs, etc.
-     *
-     * @param onTerminated
-     *   callback function to be called upon abnormal termination
+     * @param controller A reference to the Controller for launching requests, jobs, etc.
+     * @param onTerminated A callback function to be called upon abnormal termination
      *   of the task. Set it to 'nullptr' if no call back should be made.
-     *
-     * @param replicationIntervalSec
-     *   the number of seconds to wait in the end of each iteration loop before
-     *   to begin the new one.
-     *
-     * @param numReplicas
-     *   the desired number of replicas
-     *
-     * @param purge
-     *   purge excess replicas if 'true'
-     *
-     * @return
-     *   the smart pointer to a new object
+     * @param qservSyncTimeoutSec The maximum number of seconds to be waited before giving
+     *   up on the Qserv synchronization requests.
+     * @param forceQservSync Force chunk removal at worker resource collections if 'true'.
+     * @param replicationIntervalSec The number of seconds to wait in the end of each
+     *   iteration loop before to begin the new one.
+     * @param numReplicas The desired number of replicas.
+     * @param purge Purge excess replicas if 'true'.
+     * @return The smart pointer to a new object
      */
     static Ptr create(Controller::Ptr const& controller,
                       Task::AbnormalTerminationCallbackType const& onTerminated,
-                      unsigned int qservSyncTimeoutSec, unsigned int replicationIntervalSec,
-                      unsigned int numReplicas, bool purge);
+                      unsigned int qservSyncTimeoutSec, bool forceQservSync,
+                      unsigned int replicationIntervalSec, unsigned int numReplicas, bool purge);
```

Comment:
Not sure if I asked you about this already - maybe it's worth switching to `std::chrono` types for timeouts, for better type safety?

Reply:
This is a good idea. It was already discussed in the past. Iy's in my refactoring list for the R-I system. The `unsigned int` type is used in hundreds of locations across the code. Originally, this was used because the code is based on Boost ASIO which didn't accept the `std::chrono` at the time when this code was written. Boost had (and AFAIK still has its own type `boost::asio::chrono` for that). Besides, in some cases, I need to pass timeouts from the Controllers to the R-I workers via Protobuf. So, I thought that I wouldn't gain much due to these "border conditions" (type transformations between `std::chrono`/`boost::asio::chrono`/Protobuf. I should revisit the status of Boots ASIO to see if it now accepts `std::chrono` natutrally. I'm not sure how convincing this story sounds.
However, thank you for bringing this up!